File: app/src/main/java/com/innso/mobile/ui/itemViews/ItemDetailBill.java
package com.innso.mobile.ui.itemViews;

import android.content.Context;
import androidx.databinding.DataBindingUtil;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.FrameLayout;

import com.innso.mobile.R;
import com.innso.mobile.databinding.ItemGenericFinanceDetailBinding;
import com.innso.mobile.ui.bindings.GeneralBindings;
import com.innso.mobile.ui.interfaces.GenericItemView;
import com.innso.mobile.ui.models.Bill;

public class ItemDetailBill extends FrameLayout implements GenericItemView<Bill> {

    private ItemGenericFinanceDetailBinding binding;

    private Bill bill;

    public ItemDetailBill(@NonNull Context context) {
        super(context);
        init();
    }

    public ItemDetailBill(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public ItemDetailBill(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        binding = DataBindingUtil.inflate(LayoutInflater.from(getContext()), R.layout.item_generic_finance_detail, this, true);
        setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));
    }

    @Override
    public void bind(Bill data) {
        this.bill = data;
        binding.textViewDate.setText(this.bill.date);
        binding.textViewTotal.setText(this.bill.total);
        binding.textViewCustomer.setText(this.bill.customer);
        GeneralBindings.loadImage(binding.imageViewBillImage, bill.imageUrl);
    }

    @Override
    public Bill getData() {
        return bill;
    }
}

File: app/src/main/java/com/innso/mobile/ui/itemViews/ItemDetailBill.kt
package com.innso.mobile.ui.itemViews

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout

import com.innso.mobile.R
import com.innso.mobile.databinding.ItemGenericFinanceDetailBinding
import com.innso.mobile.ui.bindings.GeneralBindings
import com.innso.mobile.ui.interfaces.GenericItemView
import com.innso.mobile.ui.models.Bill
import androidx.databinding.DataBindingUtil

class ItemDetailBill : FrameLayout, GenericItemView<Bill> {

    private lateinit var binding: ItemGenericFinanceDetailBinding

    private lateinit var bill: Bill

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        binding = DataBindingUtil.inflate(LayoutInflater.from(context), R.layout.item_generic_finance_detail, this, true)
        layoutParams = FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.WRAP_CONTENT)
    }

    override fun bind(data: Bill) {
        this.bill = data
        binding.textViewDate.text = this.bill.date
        binding.textViewTotal.text = this.bill.total
        binding.textViewCustomer.text = this.bill.customer
        GeneralBindings.loadImage(binding.imageViewBillImage, bill.imageUrl)
    }

    override fun getData(): Bill? {
        return bill
    }
}
